src/main/kotlin/eu/europa/ec/eudi/sdjwt/vc/SdJwtVcVerifier.kt

```diff
@@ -140,13 +147,38 @@ class SdJwtVcVerifier(
         unverifiedSdJwt: JsonObject,
         challenge: JsonObject? = null,
     ): Result<Pair<SdJwt.Presentation<JwtAndClaims>, JwtAndClaims?>> = coroutineScope {
-        val jwtSignatureVerifier = async { jwtSignatureVerifier() }
+        val jwtSignatureVerifier = async { sdJwtVcSignatureVerifier(httpClientFactory, trust, lookup) }
         val keyBindingVerifier = KeyBindingVerifier.forSdJwtVc(challenge)
         SdJwtVerifier.verifyPresentation(jwtSignatureVerifier.await(), keyBindingVerifier, unverifiedSdJwt)
     }
 
-    private fun jwtSignatureVerifier(): JwtSignatureVerifier =
-        sdJwtVcSignatureVerifier(httpClientFactory, trust, lookup)
+    companion object {
+
+        /**
+         * Creates a new [SdJwtVcVerifier] with SD-JWT-VC Issuer Metadata resolution enabled.
+         */
+        fun usingIssuerMetadata(
+            httpClientFactory: KtorHttpClientFactory,
+        ): SdJwtVcVerifier = SdJwtVcVerifier(httpClientFactory = httpClientFactory)
+
+        /**
+         * Creates a new [SdJwtVcVerifier] with X509 Certificate trust enabled.
+         */
+        fun usingX5c(x509CertificateTrust: X509CertificateTrust): SdJwtVcVerifier = SdJwtVcVerifier(trust = x509CertificateTrust)
+
+        /**
+         * Creates a new [SdJwtVcVerifier] with DID resolution enabled.
+         */
+        fun usingDID(didLookup: LookupPublicKeysFromDIDDocument): SdJwtVcVerifier = SdJwtVcVerifier(lookup = didLookup)
+
+        /**
+         * Creates a new [SdJwtVcVerifier] with SD-JWT-VC Issuer Metadata resolution enabled, X509 Certificate trust enabled.
+         */
+        fun usingIssuerMetadataOrX5c(
```

Comment:
Please rename to `usingX5cOrIssuerMetadata`, to indicate that `x5c` takes precedence